qiskit/primitives/base_sampler.py

```diff
@@ -0,0 +1,179 @@
+        print([q.binary_probabilities() for q in result[0].quasi_dists])
+
+        # result of pqc(theta2)
+        print([q.binary_probabilities() for q in result[1].quasi_dists])
+
+        # result of pqc2(theta3)
+        print([q.binary_probabilities() for q in result[2].quasi_dists])
+
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Callable, Iterable, Sequence
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.exceptions import QiskitError
+
+from .sampler_result import SamplerResult
+
+
+class BaseSampler(ABC):
+    """Sampler base class
+
+    Base class of Sampler that calculates quasi-probabilities of bitstrings from quantum circuits.
+    """
+
+    def __init__(
+        self,
+        circuits: Iterable[QuantumCircuit],
+        parameters: Iterable[Iterable[Parameter]] | None = None,
+    ):
+        """
+        Args:
+            circuits (list[QuantumCircuit]): quantum circuits to be executed
+            parameters (list[list[Parameter]]): parameters of quantum circuits
+                Defaults to `[circ.parameters for circ in circuits]`
+
+        Raises:
+            QiskitError: for mismatch of circuits and parameters list.
+        """
+        self._circuits = tuple(circuits)
+        if parameters is None:
+            self._parameters = tuple(circ.parameters for circ in circuits)
+        else:
+            self._parameters = tuple(ParameterView(par) for par in parameters)
+            if len(self._parameters) != len(self._circuits):
+                raise QiskitError(
+                    f"Different number of parameters ({len(self._parameters)} "
+                    f"and circuits ({len(self._circuits)}"
+                )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        self.close()
+
+    @abstractmethod
+    def close(self):
+        """Close the session and free resources"""
+        ...
+
+    @property
+    def circuits(self) -> tuple[QuantumCircuit, ...]:
+        """Quantum circuits
+
+        Returns:
+            tuple[QuantumCircuit]: quantum circuits
+        """
+        return self._circuits
+
+    @property
+    def parameters(self) -> tuple[ParameterView, ...]:
+        """Parameters of quantum circuits
+
+        Returns:
+            tuple[ParameterView]: Parameter list of the quantum circuits
+        """
+        return self._parameters
+
+    @abstractmethod
+    def __call__(
+        self,
+        circuits: Sequence[int],
+        parameters: Sequence[Sequence[float]],
+        **run_options,
+    ) -> SamplerResult:
+        """Run the sampling of bitstrings.
+
+        Args:
+            circuits (list[int]): indexes of the circuits to evaluate.
+            parameters (list[list[float]]): parameters to be bound.
+            run_options: backend runtime options used for circuit execution.
```

Comment:
Docstring types don't match the type hints.  You should be ok to just remove the docstring hints - Sphinx should autopopulate them from the annotations.